services/diagnostic/sessions.go

```diff
@@ -0,0 +1,172 @@
+package diagnostic
+
+import (
+	"bytes"
+	"errors"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/influxdata/kapacitor/uuid"
+)
+
+type SessionsStore interface {
+	Create(w http.ResponseWriter, contentType string, tags []tag) *Session
+	Delete(s *Session) error
+	Each(func(*Session))
+}
+
+type sessionsStore struct {
+	mu       sync.RWMutex
+	sessions map[uuid.UUID]*Session
+}
+
+func (kv *sessionsStore) Create(w http.ResponseWriter, contentType string, tags []tag) *Session {
+	kv.mu.Lock()
+	defer kv.mu.Unlock()
+
+	flusher, ok := w.(http.Flusher)
+	var wf WriteFlusher = &noopWriteFlusher{w: w}
+	if ok {
+		wf = &httpWriteFlusher{w: w, f: flusher}
```

Comment:
You shouldn't need the type httpWriteFlusher, as w is already a writer and you have asserted that its a flusher, so you can now just type assert it to be a WriteFlusher interface. 

```
wf = w.(WriteFlusher)
```

Actually you can skip the direct flusher check and just check it its a `WriteFlusher` to begin with.

Reply:
fixed